crates/ra_db/Cargo.toml

```diff
@@ -15,4 +15,5 @@ rustc-hash = "1.1.0"
 ra_syntax = { path = "../ra_syntax" }
 ra_cfg = { path = "../ra_cfg" }
 ra_prof = { path = "../ra_prof" }
+ra_proc_macro = { path = "../ra_proc_macro" }
```

Comment:
Hm, I wonder if we can minimize the interface here...

Can we use something like this instead?

```
struct CrateData {
   proc_macros: FxHashMap<SmolStr, Arc<dyn Fn(tt::TokenTree) -> tt::TokenTree>>
}
```

Ie, define a really minimal proc macro interface directly in the `ra_db` crate, where the minimal interface is ideally an `dyn Fn()`, or maybe a named trait.

Reply:
I changed to :
```rust

#[derive(Debug, Clone)]
pub struct ProcMacro {
    pub name: SmolStr,
    pub expander: Arc<dyn TokenExpander>,
}

impl Eq for ProcMacro {}
impl PartialEq for ProcMacro { ... }

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct CrateData {
   ...
    pub proc_macro: Vec<ProcMacro>,
}
```